File: src/Frontend/properNutritionDiary-frontend/src/Features/UserMenu/Generate/GenerateMenuConfiguration.ts
export interface NutrientFilterType {
    TargetCalories?: number;
    TargetProtein?: number;
    TargetFats?: number;
    TargetCarbohydrates?: number;
    Other?: { [key: string]: number };
}

export interface SectionMenuFilter extends BaseGenerateMenuConfiguration {
    MealType: string[];
}

export interface BaseGenerateMenuConfiguration {
    NutrientFilter?: NutrientFilterType;
    Dish: string[];
}

export interface GenerateMenuConfiguration extends BaseGenerateMenuConfiguration {
    Breakfast?: SectionMenuFilter;
    Lunch?: SectionMenuFilter;
    Dinner?: SectionMenuFilter;
    Cuisine: string[];
    Diet: string[];
    Health: string[];
    DayCount: number;
}

File: src/UserMenu/ProperNutritionDiary.UserMenuApi/UserMenu/Create/GenerateMenuConfiguration.cs
namespace ProperNutritionDiary.UserMenuApi.UserMenu.Create;

public class BaseGenerateMenuConfiguration
{
    public class SectionMenuFilter : BaseGenerateMenuConfiguration
    {
        public string[] MealType { get; set; } = [];
    }

    public class NutrientFilterType
    {
        public decimal? TargetCalories { get; set; }
        public decimal? TargetProtein { get; set; }
        public decimal? TargetFats { get; set; }
        public decimal? TargetCarbohydrates { get; set; }

        public Dictionary<string, decimal> Other { get; set; } = [];
    }

    public NutrientFilterType? NutrientFilter { get; set; }
    public string[] Dish { get; set; } = [];
}

public class GenerateMenuConfiguration : BaseGenerateMenuConfiguration
{
    public SectionMenuFilter? Breakfast { get; set; }
    public SectionMenuFilter? Lunch { get; set; }
    public SectionMenuFilter? Dinner { get; set; }
    public string[] Cuisine { get; set; } = [];
    public string[] Diet { get; set; } = [];
    public string[] Health { get; set; } = [];
    public int DayCount { get; set; }
}
